File: OneCodeTeam/How to use RESTful WCF Service with HTTP requests like GetPostPutDelete [/[VB]-How to use RESTful WCF Service with HTTP requests like GetPostPutDelete [/VB/VBRESTfulWCFService/VBRESTfulWCFServiceProvider/UserService.svc.vb
'****************************** Module Header ******************************\
' Module Name:  UserService.svc.vb
' Project:      VBRESTfulWCFServiceProvider
' Copyright (c) Microsoft Corporation.
'
' WCF Service class to provide operations
'
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx.
' All other rights reserved.
'
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************/

''' <summary>
''' WCF Service class to provide operations
''' </summary>
''' <remarks></remarks>
Friend Class UserService
    Implements IUserService

#Region "Methods"

    Public Sub CreateUser(ByVal user As User) Implements IUserService.CreateUser
        user.UserObject.CreateUser(user)
    End Sub

    Public Sub DeleteUser(ByVal id As String) Implements IUserService.DeleteUser
        User.UserObject.DeleteUser(id)
    End Sub

    Public Function GetAllUsers() As System.Collections.Generic.List(Of User) Implements IUserService.GetAllUsers
        Return User.UserObject.GetAllUsers()
    End Function

    Public Sub UpdateUser(ByVal user As User) Implements IUserService.UpdateUser
        user.UserObject.UpdateUser(user)
    End Sub

#End Region

End Class

File: OneCodeTeam/How to use RESTful WCF Service with HTTP requests like GetPostPutDelete [/[C#]-How to use RESTful WCF Service with HTTP requests like GetPostPutDelete [/C#/CSRESTfulWCFService/CSRESTfulWCFServiceProvider/UserService.svc.cs
/****************************** Module Header ******************************\
Module Name:  UserService.svc.cs
Project:      CSRESTfulWCFServiceProvider
Copyright (c) Microsoft Corporation.
	 
WCF Service class to provide operations
	 
This source is subject to the Microsoft Public License.
See http://www.microsoft.com/en-us/openness/licenses.aspx.
All other rights reserved.
	 
THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System.Collections.Generic;
using CSRESTfulWCFServiceProvider.Utilities;

namespace CSRESTfulWCFServiceProvider
{
    /// <summary>
    /// WCF Service class to provide operations
    /// </summary>
    internal class UserService : IUserService
    {
        #region Methods

        /// <summary>
        /// Get all users
        /// </summary>
        /// <returns>Return user list</returns>
        public List<User> GetAllUsers()
        {
            return User.UserObject.GetAllUsers();
        }

        public void CreateUser(User user)
        {
            User.UserObject.CreateUser(user);
        }

        public void UpdateUser(User user)
        {
            User.UserObject.UpdateUser(user);
        }

        public void DeleteUser(string id)
        {
            User.UserObject.DeleteUser(id);
        }

        #endregion
    }
}
